src/daft-logical-plan/src/optimization/optimizer.rs

```diff
@@ -134,6 +134,11 @@ impl Default for OptimizerBuilder {
                     vec![Box::new(PushDownLimit::new())],
                     RuleExecutionStrategy::FixedPoint(Some(3)),
                 ),
+                // --- Simplify expressions before scans are materialized ---
+                RuleBatch::new(
+                    vec![Box::new(SimplifyExpressionsRule::new())],
+                    RuleExecutionStrategy::FixedPoint(Some(3)),
```

Comment:
Since exprs can be pretty poorly written due to generated code, etc. I'm wondering if we should do more than 3 passes.

Reply:
Gonna just use the configurable `default_max_optimizer_passes` then.